pkg/controller/history/controller_history_test.go

```diff
@@ -302,7 +302,11 @@ func TestRealHistory_CreateControllerRevision(t *testing.T) {
 		t.Fatal(err)
 	}
 	ss1Rev1.Namespace = ss1.Namespace
-	ss1Rev2, err := NewControllerRevision(ss1, parentKind, ss1.Spec.Template.Labels, rawTemplate(&ss1.Spec.Template), 2, ss1.Status.CollisionCount)
+	modTemplate := ss1.Spec.Template.DeepCopy()
+	modTemplate.Labels["foo"] = "not_bar"
+	ss1Rev2, err := NewControllerRevision(ss1, parentKind, ss1.Spec.Template.Labels, rawTemplate(modTemplate), 2, ss1.Status.CollisionCount)
+	ss1Rev2.Name = ss1Rev1.Name
```

Comment:
By modifying labels of `modTemplate`, the hash for populating name and labels fields will change, so I think it is incorrect to set `ss1Rev2.Name = ss1Rev1.Name` (same with `.Labels` below).

Reply:
The labels are being manipulated in this way in order to set up the scenario where there are two controller revisions with the same name and hash label, but which are not identical. This is done by creating a new controller revision which is slightly different than an existing one (change the value of a label) and then give it the same name and label as the existing one. This will create a controller revision where the name and hash are "incorrect" for the content, but it works for setting up the data for the test. Normally this would be the result of a hash collision, we don't have a way to create those in the test.

Reply:
Noted. As this is a unit test, I think this can be made clearer by creating the 2nd `ControllerRevision` (with the same name and labels) before modifying it's labels.

Reply:
As sync by person, `ControllerRevision` is immutable once created, so this "hack" is necessary. I will LGTM again once comments are added. Thanks for the work!